### DISCUSSION TITLE
Adding Application Insights to .NET MAUI Application

### DISCUSSION THREAD

**Discussion by giscus** [VladislavAntonyuk/vladislavantonyuk.github.io#108] [Announcements]
# Adding Application Insights to .NET MAUI Application

This article provides an in-depth exploration into how you can integrate Microsoft's Application Insights into your .NET MAUI application. A comparative study between Microsoft AppCenter and Application Insights is also highlighted, demystifying the inherent advantage of Application Insights in application management and analytics.

https://vladislavantonyuk.github.io/articles/Adding-Application-Insights-to-.NET-MAUI-Application/

**Comment by iPaul-777** [comment]
The constructor is taking in a parameter of type ILogger<MyService> and assigning it to the private variable of the same name but I don't see anywhere where this constructor is called with that argument.  If you were to create a default constructor then logger will not be defined and it's obviously going to throw an exception.  Where does logged come from?

**Reply by VladislavAntonyuk** [reply]
From DI. See MauiProgram where we register logger

**Reply by iPaul-777** [reply]
Do you have a MauiProgram.cs example that initializes MyService() as shown above?  
This code does not initialize MyService():
builder.Logging.AddApplicationInsights(configuration =>
{
    configuration.TelemetryInitializers.Add(new ApplicationInitializer());
    configuration.ConnectionString = "YOUR-CONNECTION-STRING;
}, options =>
{
    options.IncludeScopes = true;
});

**Reply by VladislavAntonyuk** [reply]
Of course, this code registers ApplicationInsights logger. MyService can be registered as:
```
builder.Services.AddSingleton<MyService>();
```

The full sample is available. See the link in the bottom of the article.
